### PR TITLE
[GEOS-9539]: avoid fixing correct urls in URLKvpParser

### PR DESCRIPTION
**Pull request by mbarto**
<Include a few sentences describing the overall goals for this Pull Request>
URLKvpParser fixes every url, also non broken ones, introducing a wrong encoding in some cases.
This fix tries to use the url right away, and fix it only if it is not valid.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [x] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [x] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [ ] Commits changing the REST API, or any configuration object, should check if the REST API docs (Swagger YAML files and classic documentation) need to be updated.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

File: src/ows/src/test/java/org/geoserver/ows/kvp/URLKvpParserTest.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+public class URLKvpParserTest {
+    URLKvpParser parser;
+
+    @Before
+    public void setUp() {
+        parser = new URLKvpParser("url");
+    }
+
+    @Test
+    public void testValidUrl() {
```
Comment (bradh) [inline src/ows/src/test/java/org/geoserver/ows/kvp/URLKvpParserTest.java]: Is there already a test for invalid URL?

Comment (mbarto) [reply src/ows/src/test/java/org/geoserver/ows/kvp/URLKvpParserTest.java]: @bradh the code was not tested before, and it's not even clear to me why the fixUrl was introduced, because it was breaking a valid url, that now is instead working properly.
To give you some context: this parser is only used at the moment to parse WMS SLD parameter.
When parsing a url with a complex querystring, used to produce a dynamic style with the sldService extension, the fixed url was indeed broken, while with the new code path it works perfectly.
The doubt was if we wanted to remove fixURL completely, but not understanding its real purpose we decided to use it as a fallback.
Long story short: I couldn't find a test case for an invalid url, because I couldn't imagine the purpose of the original code.

Comment (aaime) [reply src/ows/src/test/java/org/geoserver/ows/kvp/URLKvpParserTest.java]: My guess is that lots of existing tests in GeoServer are not properly url-encoding the urls, so if you skip the "fixurl" part they will start failing. But it's just a guess.

Comment (aaime) [reply src/ows/src/test/java/org/geoserver/ows/kvp/URLKvpParserTest.java]: @jdeolive I know it was from many many years ago, but do you remember by any chance why the "on the fly url fixing" was introduced in the URLKvpParser, to start with? Seems like it came in with the first implementation of KVP parsers.

Comment (aaime) [reply src/ows/src/test/java/org/geoserver/ows/kvp/URLKvpParserTest.java]: Actually tried it out, removing the fixurl part, letting it throw an exception, and the build still works. Tried to put together a URL that would throw a fixable MalformedURLException, but could not... maybe the behavior of Java own URL class changed over time?
In any case, I'd keep the code as is for safety, since we don't know exactly what it fixes, and move on.
